lib/cartopy/mpl/geoaxes.py

```diff
@@ -784,6 +796,187 @@ def stock_img(self, name='ne_shaded'):
+                        latitude start, latitude end].
+
+                       e.g. [-30.0, 10.0, -30.0, 10.0] for Ascension Island
+                       or [167.0, 193.0, 47.0, 68.0] to cross the date line.
+
+            * cache - logical flag as to whether or not to cache the loaded
+                      images into memory. The images are stored before the
+                      extent is used.
+        """
+        # read in the user's background image directory:
+        if len(_USER_BG_IMGS) == 0:
+            self.read_user_backgound_images()
+        import os
+        bgdir = os.getenv('CARTOPY_USER_BACKGROUNDS')
+        if bgdir is None:
+            bgdir = os.path.join(config["repo_data_dir"],
+                                 'raster', 'natural_earth')
+        # now get the filename we want to use:
+        try:
+            fname = _USER_BG_IMGS[name][resln]
+        except KeyError:
+            err_str = 'Image name "{}" and resolution "{}"'.format(name, resln)
+            err_str += ' are not present in the user background image metadata'
+            err_str += ' in directory "{}"'.format(bgdir)
+            raise ValueError(err_str)
+        # Now obtain the image data from file or cache:
+        fpath = os.path.join(bgdir, fname)
+        if cache:
+            if fname in _BACKG_IMG_CACHE.keys():
+                img = _BACKG_IMG_CACHE[fname]
+            else:
+                img = imread(fpath)
+                _BACKG_IMG_CACHE[fname] = img
+        else:
+            img = imread(fpath)
+        if len(img.shape) == 2:
+            # greyscale images are only 2-dimensional, so need replicating
+            # to 3 colour channels:
+            img = np.repeat(img[:, :, np.newaxis], 3, axis=2)
+
+        # now get the projection from the metadata:
+        if _USER_BG_IMGS[name]['__projection__'] == 'PlateCarree':
+            # currently only PlateCarree is defined:
+            source_proj = ccrs.PlateCarree()
+        else:
+            raise NotImplementedError('Background image projection undefined')
+
+        if extent is None:
+            # not specifying an extent, so return all of it:
+            return self.imshow(img, origin='upper',
+                               transform=source_proj,
+                               extent=[-180, 180, -90, 90])
+        else:
+            # return only a subset of the image:
+            # set up coordinate arrays:
+            d_lat = 180.0 / img.shape[0]
+            d_lon = 360.0 / img.shape[1]
+            # latitude starts at 90N for this image:
+            lat_pts = (np.arange(img.shape[0]) * -d_lat - (d_lat / 2.0)) + 90.0
+            lon_pts = (np.arange(img.shape[1]) * d_lon + (d_lon / 2.0)) - 180.0
+
+            # which points are in range:
+            lat_in_range = np.logical_and(lat_pts >= extent[2],
+                                          lat_pts <= extent[3])
+            if extent[0] < 180 and extent[1] > 180:
```

Comment:
That bit of logic catches extents that cross the dateline. An extent[1] that's negative doens't span the dateline so works in the else block.